### PR TITLE
feat: add arm64 support and use a non-root path

### LINKED ISSUE #2: Download the Coder binary to the workflow workspace folder
Currently the action tries to run download the coder binary and store it in `/usr/local/bin/coder` which requires the runner to run with root priviledges. In some systems that might not be possible (either very strict IT policies, or very closed down environment). 

Suggestion: Could we change the path to the path of the workflow workspace (where the runner should have permissions).

```
runs:
  using: "composite"
  steps:
    - name: Download Coder binary from access URL
      run: curl -fsSL $CODER_URL/bin/coder-linux-amd64 -o /usr/local/bin/coder && chmod +x /usr/local/bin/coder
      shell: bash
      env:
        CODER_URL: ${{ inputs.access_url }}
```

### PR DESCRIPTION
**Pull request by matifali**
- add tests
- add arm64 support
- ensure binaries are installed in the local bin directory to support self-hosted runners without sudo or root access.

Closes #2

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -17,15 +17,23 @@ inputs:
...
-      run: curl -fsSL $CODER_URL/bin/coder-linux-amd64 -o /usr/local/bin/coder && chmod +x /usr/local/bin/coder
+    - name: Download and install Coder binary
+      run: |
+        if [ "${{ runner.arch }}" == "ARM64" ]; then
+          ARCH="arm64"
+        else
+          ARCH="amd64"
+        fi
+        mkdir -p $HOME/.local/bin
+        curl -fsSL ${{ inputs.access_url }}/bin/coder-linux-${ARCH} -o $HOME/.local/bin/coder
```
Comment (mafredri) [inline action.yaml]: Any reason to change this from env to template value? Env is safer with regards to shell escaping. I.e. `"${CODER_URL}"` will definitely work, whereas `"${{inputs.access_url}}"` could contain `"` and break the script.

File: action.yaml [hunk 2]
```diff
@@ -17,15 +17,23 @@ inputs:
...
-    - name: Download Coder binary from access URL
-      run: curl -fsSL $CODER_URL/bin/coder-linux-amd64 -o /usr/local/bin/coder && chmod +x /usr/local/bin/coder
+    - name: Download and install Coder binary
+      run: |
+        if [ "${{ runner.arch }}" == "ARM64" ]; then
+          ARCH="arm64"
+        else
+          ARCH="amd64"
+        fi
+        mkdir -p $HOME/.local/bin
```
Comment (mafredri) [inline action.yaml]: Good practice to quote all variables, i.e.
```suggestion
        mkdir -p "$HOME"/.local/bin
```

**Review by mafredri — APPROVED** [review]
Minor nits but otherwise LGTM 👍🏻

File: action.yaml (hunk 1 above)
Comment (matifali) [reply action.yaml]: My thinking is that the ENV is not available at this stage so we should use the input directly.

Comment (mafredri) [reply action.yaml]: Hmm, why isn't it available if set via `env:` in yaml? If that's the case then sure.

Comment (matifali) [reply action.yaml]: addressed.

File: action.yaml [hunk 3]
```diff
@@ -17,8 +17,18 @@ inputs:
...
-      run: curl -fsSL $CODER_URL/bin/coder-linux-amd64 -o /usr/local/bin/coder && chmod +x /usr/local/bin/coder
+    - name: Download and install Coder binary
+      run: |
+        if [ "${{ runner.arch }}" == "ARM64" ]; then
+          ARCH="arm64"
+        else
+          ARCH="amd64"
+        fi
+        mkdir -p "$HOME"/.local/bin
+        curl -fsSL "$CODER_URL"/bin/coder-linux-${ARCH} -o "$HOME"/.local/bin/coder
```
Comment (mafredri) [inline action.yaml]: Should we add `--compressed` to speed up the download?

**Review by mafredri — APPROVED** [review]
Had one more idea, but LGTM!

File: action.yaml (hunk 3 above)
Comment (matifali) [reply action.yaml]: Yeah good idea. Although GitHub runners are fast but as we are fetching from Coder server it can be on a slow network.

**Pull request merged**